homeassistant/components/homekit_controller/__init__.py

```diff
@@ -232,5 +232,9 @@ def get_characteristic_types(self):
 
 async def async_remove_entry(hass, entry):
     """Cleanup caches before removing config entry."""
+    if "AccessoryPairingID" not in entry.data:
```

Comment:
I think that we should instead not call `async_remove_entry` at all.

Reply:
Yeah I did wonder about that too -  will go and try and find out how then :)

Reply:
Or, and this is a better idea. We can leverage `async_remove_entry` as a way for integrations to try to rediscover and start the config flow !